gin.go

```diff
@@ -745,6 +769,14 @@ func serveError(c *Context, code int, defaultMessage []byte) {
 	c.writermem.WriteHeaderNow()
 }
 
+func addOrRemoveTrailingSlash(p string) string {
+	ret := p + "/"
+	if length := len(p); length > 1 && p[length-1] == '/' {
+		ret = p[:length-1]
+	}
+	return ret
```

Comment:
```suggestion
	if length := len(p); length > 1 && p[length-1] == '/' {
		return p[:length-1]
	}
	return p + "/"
```

No need variable. just return like before your changes